test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
                                  (Thread/sleep 500)
                                  (recur (dec times)))))))))))
+
+
+
+(deftest ^:integration revoke-compile-master-test
+  (testing "Revoking compile master certificate"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
+        (testing "should update infrastructure CRL"
+          (let [ca-cert (bootstrap/get-ca-cert-for-running-server)
+                subject "dummy-compile-master"
+                ;; In one of the other unit tests I noticed the localhost cert is
+                ;; being revoked. If that does not affect things we could do that instead of 
+                ;; creating a new one. But wondering it actually may have impact on other 
+                ;; tests since this is integration
+                cm-cert (bootstrap/get-cert-signed-by-ca-for-running-server
+                             ca-cert
+                             subject)
+                 ;; TODO - Extract serial number of abvoe cert. Is there an easy way
+                 ;; to do it here or can we make assumption on the expected serial number
+                 ;;
+
+                 ;; Q: Do we need to do something to explicitly moved the certificate
+                 ;; to CA's signed folder
+                 cert-status-request (fn [action]
+                                       (http-client/put
+                                         (str "https://localhost:8140/"
+                                              "puppet-ca/v1/certificate_status/"
+                                              subject)
+                                         (merge options
+                                                {:body (str "{\"desired_state\": \""
+                                                            action
+                                                            "\"}")
+                                                 :headers {"content-type"
+                                                           "application/json"}}))) ]
+            (testing "Infra CRL should contain the revoked cert"
+               (let [revoke-response (cert-status-request "revoked")]
+                 ;; If the revocation was successful we want to check the infra CRL contains above certificate
+                 ;; Can I just change the below to have an AND check as I have attempted below
+                 ;; Q: How do I access CaSettings or just certificate-authority settings here to look up infra-crl-path?  
```

Comment:
Couple different levels of testing here. Since you're going through the HTTP API I would expect that you'd go through it as well to validate that the correct CRL was returned when trying to download it. Otherwise you could also call the CA function `revoke-existing-cert!` that's called by the HTTP API and test at a lower level. You can also pass in a custom path to the infra-crl, and serial-file right? If you're explicitly setting paths, I don't think it's inappropriate to then read those paths at the end of the test.

Reply:
yeah perhaps we should test the downloading part. And checking that the revoked cert shows up in the CRL will ensure 'correctness' (which am trying to do in the current version of the test). Btw, I have not yet changed the place where the http get crl API to return the infra (that will then allow testing the downloads correct crl). We need a new config setting to control whether to return full crl or infra. Should that be part of this same PR?   

Coming back to my question of - how do I access the CaSettings configuration? Basically I want to access :infra-crl-path from "settings" (what would the settings be in this case)?

Reply:
That value isn't readily available that I know of